website/server/libs/stripePayments.js

```diff
@@ -226,20 +227,27 @@ api.cancelSubscription = async function cancelSubscription (options, stripeInc)
   if (!customerId) throw new NotAuthorized(i18n.t('missingSubscription'));
 
   // @TODO: Handle error response
-  let customer = await stripeApi.customers.retrieve(customerId);
+  let customer = await stripeApi.customers.retrieve(customerId).catch(function errorCatch (err) {
+    return err;
```

Comment:
I'd like to log it somewhere, but do we have a good way of doing that at present? (console logging is bad for Lint)

By catching the error, though, if the customer ID doesn't exist on the payment processor end--such as when the subscription has been remotely canceled due to failed payments--we will go ahead and cancel the subscription on the Habitica side. That fixes users seeing a bare API error when they try to cancel a dead Stripe sub, and helps us prune unpaid-for subscriptions pending better Webhook handling. @TheHollidayInn may have further insight!